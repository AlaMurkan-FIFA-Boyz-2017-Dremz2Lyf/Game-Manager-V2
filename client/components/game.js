import React, { Component } from 'react';
import { connect } from 'react-redux';

import { update } from '../actions';

import {
  Modal,
  ListGroupItem,
  Button,
  Row,
  Col
} from 'react-bootstrap';

import GameDetails from './game_details';
import GameForm from './game_form';

export class Game extends Component {
  constructor() {
    super();
    this.state = {
      showModal: false
    };
  }

  open() {
    this.setState({
      showModal: true
    });
  }

  close() {
    this.setState({
      showModal: false
    });
  }

  handleSubmit(values) {
    // Pull the game, the tournaments, and the update action creator from props.
    let { game, tournaments, update } = this.props;

    // then grab the id, tournament id and p1Score from the game.
    let { id, tournament, p1Score} = game;

    // lastly, grab the gamesPlayed from the tournament corresponding to this game.
    let { gamesPlayed } = tournaments[tournament];

    // Because both scores are required to update a game, we can check if the game has been played with just one of the scores.
    if (p1Score === null) {
      // If this is the first time this game is being updated, we increment the game count, and update the tournament.
      ++gamesPlayed;
      update('tournaments', {id: tournament, gamesPlayed});
    }

    // Update the game in the database with the values from the form and a status of finished.
    update('games', {id, ...values, status: 'finished'});
    this.setState({
      showModal: false
    });
  }

  render() {
    let { game } = this.props;
    let { players } = this.props;

    return (
      <ListGroupItem onClick={this.open.bind(this)}>
        <GameDetails
          game={game}
          player1={players[game.p1]}
          player2={players[game.p2]}
        />
        <Modal show={this.state.showModal} onHide={this.close.bind(this)}>
          <Modal.Header>
            <Modal.Title>Finish this Game!</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Row>
              <Col xs={3}>
              </Col>
              <Col xs={2}>
                <h4>{players[game.p1].username}</h4>
              </Col>
              <Col xs={2}>
                <h4>{players[game.p2].username}</h4>
              </Col>
              <Col xs={5}>
              </Col>
            </Row>
            <Row>
              <Col xs={12}>
                <GameForm
                  onSubmit={this.handleSubmit.bind(this)}
                  initialValues={game}
                />
              </Col>
            </Row>
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={this.close.bind(this)}>Cancel</Button>
          </Modal.Footer>
        </Modal>
      </ListGroupItem>
    );
  }
}

Game.propTypes = {
  game: React.PropTypes.object.isRequired,
  players: React.PropTypes.object.isRequired
};

export const mapStateToProps = ({ data }) => ({
  players: data.players,
  tournaments: data.tournaments
});

export default connect(mapStateToProps, { update })(Game);
