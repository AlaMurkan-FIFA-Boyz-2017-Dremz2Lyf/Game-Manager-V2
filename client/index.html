<html>
<head>
  <title>Game Manager</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans|Oswald|PT+Sans">
</head>
<body>
  <div id="app"></div>
</body>
</html>
