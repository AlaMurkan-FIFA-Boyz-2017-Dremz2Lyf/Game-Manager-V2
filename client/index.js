// Dependancies
import React from 'react';
import ReactDOM from 'react-dom';
import { ProviderÂ } from 'react-redux';

// css
import '../styles/styles.css';

// Bring in our router
import Routes from './router';

// Grab our root reducer and our store from their files
import rootReducer from './reducers/index.js';
import { store } from './store';

// Found this... Needed for hot module reloading.
if (module.hot) {
  module.hot.accept();
}

ReactDOM.render(
  <Provider store={store}>
    <Routes />
  </Provider>,
  document.getElementById('app')
);
