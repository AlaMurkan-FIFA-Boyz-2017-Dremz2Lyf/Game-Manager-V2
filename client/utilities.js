/**
  * normalize
  * @param {array} list - an array of objects to be normalized into an object
*/
export const normalize = (list) => {

  return list.reduce((normalized, item) => {
    normalized[item.id] = item;
    return normalized;
  }, {});

};

/**
  * applyAll - apply all uses a bit of tricky programming here to return a new state adding to only the specific substate we need.
  * @param {object} state - the current state
  * @param {object} action - the action coming through, it's payload must have a stateKey and the data corresponding to that statekey.
*/
export const applyAll = (state, action) => {
  const { stateKey, data } = action.payload;

  return {...state, [stateKey]: data};
};

/**
  * percentPlayed - basic math here.
  * @param {number} gamesPlayed - pretty self explanatory
  * @param {number} totalGames - pretty self explanatory
*/
export const percentPlayed = (gamesPlayed, totalGames) => {
  return Math.floor((gamesPlayed / totalGames) * 100);
};

/**
  * getValidationState - will give us the correct bootstrap class for UI responsiveness
  * @param {bool} valid - passed from redux-form, lets us know if the form is valid
  * @param {bool} pristine - from redux-form, lets us know if the form has been touched & edited
*/
export const getValidationState = (valid, pristine) => pristine ? null : (valid ? 'success' : 'error' );

/**
  * required - another validation check for redux-form
  * @param {string} value - from redux-form, returns an alert if the user has not entered required info. Also deals with the number 0 as that is falsey value. 
*/
export const required = value => value === 0 || value ? undefined : 'Required';

/**
  * possValidation - validation specific to the possession field
  * @param {string} value - the input from the form
  * @param {string} allValues - all the inputs from the form
*/
export const possValidation = (value, allValues) => {
  const { p1Poss, p2Poss } = allValues;

  if (!p1Poss && !p2Poss) {
    return 'Possession is required.';
  } else if (Number(p1Poss) + Number(p2Poss) !== 100) {
    return 'Possession should add up to 100%!';
  }
  return undefined;
};

/**
  * helpMessage - renders the correct help message based on the field
  * @param {string} field
*/
export const helpMessage = (field) => {
  let message;

  if (field === 'Score') {
    message = `${field} is required! I mean.. come on!`;
  } else {
    message = `${field} will default to 0 if not given.`;
  }
  return message;
};

/**
  * buildTable
  * @param {object} games - a normalized games object to build the table from
  * @param {object} players - another normalized object to reference the player names.
*/
export const buildTable = (games, players) => {

  let trackedStats = {
    wins: 0, losses: 0, draws: 0, goalsFor: 0, goalsAgainst: 0, shots: 0, onGoal: 0, reds: 0, yellows: 0, poss: null, points: 0, goalDiff: 0
  };

  return Object.values(games).reduce((table, game) => {

    let { p1, p2 } = game;

    table[p1] = table[p1] ? table[p1] : { ...trackedStats, id: p1, username: players[p1].username};
    table[p2] = table[p2] ? table[p2] : { ...trackedStats, id: p2, username: players[p2].username};

    return applyGame(table, game);
  }, {});
};


// I extracted this out of the reduce in build table because of all the action that needs to happen.
// There might be a more programmatic way to do this. Potentially mapping the game objects from the database into nested objects for each player in the game.... that would, in theory, allow us to just loop through each player's object from the game and apply them to the correct player.

/**
  * applyGame - used in buildTable to apply the game to the table
  * @param {object} table - the current table
  * @param {object} game - the current game to be applied
*/
export const applyGame = (table, game) => {

  let { p1, p2, p1Score, p2Score, p1Shots, p2Shots, p1Poss, p2Poss, p1OnGoal, p2OnGoal, p1Reds, p1Yellows, p2Reds, p2Yellows } = game;


  let player1 = table[p1];
  let player2 = table[p2];

  if (p1Score && p2Score) {

    p1Score === p2Score ? (
      player1.draws++,
      player2.draws++
    ) : p1Score > p2Score ? (
      player1.wins++,
      player2.losses++
    ) : (
      player2.wins++,
      player1.losses++
    );
    player1.goalsFor += p1Score;
    player2.goalsFor += p2Score;
    player1.goalsAgainst += p2Score;
    player2.goalsAgainst += p1Score;
    player1.shots += p1Shots;
    player2.shots += p2Shots;
    player1.onGoal += p1OnGoal;
    player2.onGoal += p2OnGoal;
    player1.reds += p1Reds;
    player2.reds += p2Reds;
    player1.yellows += p1Yellows;
    player2.yellows += p2Yellows;
    player1.poss = player1.poss === null ? p1Poss : (player1.poss + p1Poss) / (player1.wins + player1.losses + player1.draws);
    player2.poss = player2.poss === null ? p2Poss : (player1.poss + p1Poss) / (player1.wins + player1.losses + player1.draws);
    player1.points = (player1.wins * 3 + player1.draws);
    player2.points = (player2.wins * 3 + player2.draws);
    player1.goalDiff = (player1.goalsFor - player1.goalsAgainst);
    player2.goalDiff = (player2.goalsFor - player2.goalsAgainst);
  }
  return table;
};


// NOTE: Currently not being used:
  // Need to figure out how to get the react-bootstrap - tooltips to work.
  // Seemingly I need to build a component wrapped in a <th>... but the first attempt didn't work.
export const mapHeaders = (header) => {
  switch (header) {
  case 'P':
    return 'Points';
  case 'W':
    return 'Wins';
  case 'L':
    return 'Losses';
  case 'D':
    return 'Draws';
  case 'GF':
    return 'Goals For';
  case 'GA':
    return 'Goals Against';
  case 'GD':
    return 'Goal Differencial';
  case 'S':
    return 'Shots';
  case 'OT':
    return 'Shots On Target';
  case 'Y':
    return 'Yellows';
  case 'R':
    return 'Reds';
  case 'Po':
    return 'Possession';
  default:
    return 'Player';
  }
};
