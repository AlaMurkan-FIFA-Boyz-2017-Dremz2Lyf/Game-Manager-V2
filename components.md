 Component | State | Actions
---|---|---
 Home | None? | None?
 AllTimeStats | {currentList: ongoing/finished/new} | None?
