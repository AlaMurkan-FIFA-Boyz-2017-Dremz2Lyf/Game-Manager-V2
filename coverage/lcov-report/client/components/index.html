<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> client/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>95/95</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>25/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>51/51</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>91/91</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="create_player.js"><a href="create_player.js.html">create_player.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="create_tournament.js"><a href="create_tournament.js.html">create_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="25" class="abs high">25/25</td>
	</tr>

<tr>
	<td class="file high" data-value="game.js"><a href="game.js.html">game.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="games.js"><a href="games.js.html">games.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="home.js"><a href="home.js.html">home.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="play_tournament.js"><a href="play_tournament.js.html">play_tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="player_form.js"><a href="player_form.js.html">player_form.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="15" class="abs high">15/15</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list.js"><a href="player_list.js.html">player_list.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="player_list_item.js"><a href="player_list_item.js.html">player_list_item.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="tournament.js"><a href="tournament.js.html">tournament.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="tournaments.js"><a href="tournaments.js.html">tournaments.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Mar 10 2017 17:46:58 GMT-0600 (CST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
