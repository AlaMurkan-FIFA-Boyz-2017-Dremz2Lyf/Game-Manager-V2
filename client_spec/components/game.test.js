import React from 'react';

import { Game, mapStateToProps } from '../../client/components/game';
import { normalize } from '../../client/utilities';

describe('Game Component', () => {
  let mockUpdate = jest.fn();
  let mockGame = mockData.games.slice(1, 2)[0];
  let wrapper = shallow(
    <Game
      players={normalize(mockData.players)}
      game={mockGame}
      tournaments={mockData.tournaments.slice(0)}
      update={mockUpdate}
    />
  );

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


  test('should open the game modal when clicked', () => {
    expect(wrapper.state('showModal')).toBe(false);
    expect(wrapper).toMatchSnapshot();

    wrapper.find('ListGroupItem').simulate('click');

    expect(wrapper.state('showModal')).toBe(true);
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({showModal: false});
  });

  test('should close the modal on "hide"', () => {
    wrapper.setState({showModal: true});

    let modal = wrapper.find('Modal');
    expect(modal.length).toBe(1);

    modal.simulate('hide');
    expect(wrapper.state('showModal')).toBe(false);
  });

  test('should mapStateToProps', () => {
    let state = {
      data: {
        players: {}
      }
    };
    let props = mapStateToProps(state);

    expect(props).toEqual({players: {}});
  });

  test('should handle the submit properly', () => {
    // Open the modal so the form is there to actually submit
    wrapper.setState({showModal: true});

    // Find the ReduxForm and simulate a submit event on it.
    wrapper.find('ReduxForm').simulate('submit');

    // expect the mockUpdate to have been called
    expect(mockUpdate).toHaveBeenCalled();
    // expect the correct values to have been passed to the mock
      // we do not simulate any change events for input, because those are handled by redux-form, and we trust that library has done the testing of that.
    expect(mockUpdate.mock.calls[0]).toEqual(['tournaments', {id: 1, gamesPlayed: 2}]);
    expect(mockUpdate.mock.calls[1]).toEqual(['games', {id: 2, status: 'finished'}]);
    // and finally we expect the modal to have been hidden.
    expect(wrapper.state().showModal).toBe(false);

    // clear up the mock for further tests
    mockUpdate.mockClear();
  });

  test('should not update gamesPlayed if the game has already been submitted', () => {
    // again, open the model
    wrapper.setState({showModal: true});

    // simulate that our game has already been played
    mockGame.p1Score = 1;
    mockGame.p2Score = 3;

    wrapper.setProps({game: mockGame});

    // and simulate our submit
    wrapper.find('ReduxForm').simulate('submit');
    expect(mockUpdate.mock.calls[0]).toEqual(['games', {id: 2, status: 'finished'}]);

    // clear up the mock for further tests
    mockUpdate.mockClear();
  });

});
