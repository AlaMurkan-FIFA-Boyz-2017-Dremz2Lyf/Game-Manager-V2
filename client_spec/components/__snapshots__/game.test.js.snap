exports[`Game Component should match the snapshot 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  className="even"
  listItem={false}
  onClick={[Function]}>
  <GameDetails
    game={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 2,
        "p1": 1,
        "p1OnGoal": null,
        "p1PassAcc": null,
        "p1Poss": null,
        "p1Reds": 0,
        "p1Score": null,
        "p1Shots": null,
        "p1Yellows": 0,
        "p2": 3,
        "p2OnGoal": null,
        "p2PassAcc": null,
        "p2Poss": null,
        "p2Reds": 0,
        "p2Score": null,
        "p2Shots": null,
        "p2Yellows": 0,
        "status": "created",
        "tournament": 1,
        "updatedAt": null,
      }
    }
    player1={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 1,
        "isTeam": 0,
        "updatedAt": null,
        "username": "Alice",
      }
    }
    player2={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 3,
        "isTeam": 0,
        "updatedAt": null,
        "username": "Bob",
      }
    } />
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "modals": Array [],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}>
    <ModalHeader
      aria-label="Close"
      bsClass="modal-header"
      closeButton={false}>
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4">
        Finish this Game!
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div">
      <Row
        bsClass="row"
        componentClass="div">
        <Col
          bsClass="col"
          componentClass="div"
          xs={3} />
        <Col
          bsClass="col"
          componentClass="div"
          xs={2}>
          <h4>
            Alice
          </h4>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          xs={2}>
          <h4>
            Bob
          </h4>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          xs={5} />
      </Row>
      <Row
        bsClass="row"
        componentClass="div">
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}>
          <ReduxForm
            initialValues={
              Object {
                "createdAt": "2016-12-25T19:31:48",
                "id": 2,
                "p1": 1,
                "p1OnGoal": null,
                "p1PassAcc": null,
                "p1Poss": null,
                "p1Reds": 0,
                "p1Score": null,
                "p1Shots": null,
                "p1Yellows": 0,
                "p2": 3,
                "p2OnGoal": null,
                "p2PassAcc": null,
                "p2Poss": null,
                "p2Reds": 0,
                "p2Score": null,
                "p2Shots": null,
                "p2Yellows": 0,
                "status": "created",
                "tournament": 1,
                "updatedAt": null,
              }
            }
            onSubmit={[Function]} />
        </Col>
      </Row>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div">
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}>
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</ListGroupItem>
`;

exports[`Game Component should open the game modal when clicked 1`] = `
<ListGroupItem
  bsClass="list-group-item"
  className="even"
  listItem={false}
  onClick={[Function]}>
  <GameDetails
    game={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 2,
        "p1": 1,
        "p1OnGoal": null,
        "p1PassAcc": null,
        "p1Poss": null,
        "p1Reds": 0,
        "p1Score": null,
        "p1Shots": null,
        "p1Yellows": 0,
        "p2": 3,
        "p2OnGoal": null,
        "p2PassAcc": null,
        "p2Poss": null,
        "p2Reds": 0,
        "p2Score": null,
        "p2Shots": null,
        "p2Yellows": 0,
        "status": "created",
        "tournament": 1,
        "updatedAt": null,
      }
    }
    player1={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 1,
        "isTeam": 0,
        "updatedAt": null,
        "username": "Alice",
      }
    }
    player2={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 3,
        "isTeam": 0,
        "updatedAt": null,
        "username": "Bob",
      }
    } />
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "modals": Array [],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}>
    <ModalHeader
      aria-label="Close"
      bsClass="modal-header"
      closeButton={false}>
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4">
        Finish this Game!
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div">
      <Row
        bsClass="row"
        componentClass="div">
        <Col
          bsClass="col"
          componentClass="div"
          xs={3} />
        <Col
          bsClass="col"
          componentClass="div"
          xs={2}>
          <h4>
            Alice
          </h4>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          xs={2}>
          <h4>
            Bob
          </h4>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          xs={5} />
      </Row>
      <Row
        bsClass="row"
        componentClass="div">
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}>
          <ReduxForm
            initialValues={
              Object {
                "createdAt": "2016-12-25T19:31:48",
                "id": 2,
                "p1": 1,
                "p1OnGoal": null,
                "p1PassAcc": null,
                "p1Poss": null,
                "p1Reds": 0,
                "p1Score": null,
                "p1Shots": null,
                "p1Yellows": 0,
                "p2": 3,
                "p2OnGoal": null,
                "p2PassAcc": null,
                "p2Poss": null,
                "p2Reds": 0,
                "p2Score": null,
                "p2Shots": null,
                "p2Yellows": 0,
                "status": "created",
                "tournament": 1,
                "updatedAt": null,
              }
            }
            onSubmit={[Function]} />
        </Col>
      </Row>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div">
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}>
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</ListGroupItem>
`;

exports[`Game Component should open the game modal when clicked 2`] = `
<ListGroupItem
  bsClass="list-group-item"
  className="even"
  listItem={false}
  onClick={[Function]}>
  <GameDetails
    game={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 2,
        "p1": 1,
        "p1OnGoal": null,
        "p1PassAcc": null,
        "p1Poss": null,
        "p1Reds": 0,
        "p1Score": null,
        "p1Shots": null,
        "p1Yellows": 0,
        "p2": 3,
        "p2OnGoal": null,
        "p2PassAcc": null,
        "p2Poss": null,
        "p2Reds": 0,
        "p2Score": null,
        "p2Shots": null,
        "p2Yellows": 0,
        "status": "created",
        "tournament": 1,
        "updatedAt": null,
      }
    }
    player1={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 1,
        "isTeam": 0,
        "updatedAt": null,
        "username": "Alice",
      }
    }
    player2={
      Object {
        "createdAt": "2016-12-25T19:31:48",
        "id": 3,
        "isTeam": 0,
        "updatedAt": null,
        "username": "Bob",
      }
    } />
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "modals": Array [],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}>
    <ModalHeader
      aria-label="Close"
      bsClass="modal-header"
      closeButton={false}>
      <ModalTitle
        bsClass="modal-title"
        componentClass="h4">
        Finish this Game!
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      componentClass="div">
      <Row
        bsClass="row"
        componentClass="div">
        <Col
          bsClass="col"
          componentClass="div"
          xs={3} />
        <Col
          bsClass="col"
          componentClass="div"
          xs={2}>
          <h4>
            Alice
          </h4>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          xs={2}>
          <h4>
            Bob
          </h4>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          xs={5} />
      </Row>
      <Row
        bsClass="row"
        componentClass="div">
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}>
          <ReduxForm
            initialValues={
              Object {
                "createdAt": "2016-12-25T19:31:48",
                "id": 2,
                "p1": 1,
                "p1OnGoal": null,
                "p1PassAcc": null,
                "p1Poss": null,
                "p1Reds": 0,
                "p1Score": null,
                "p1Shots": null,
                "p1Yellows": 0,
                "p2": 3,
                "p2OnGoal": null,
                "p2PassAcc": null,
                "p2Poss": null,
                "p2Reds": 0,
                "p2Score": null,
                "p2Shots": null,
                "p2Yellows": 0,
                "status": "created",
                "tournament": 1,
                "updatedAt": null,
              }
            }
            onSubmit={[Function]} />
        </Col>
      </Row>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div">
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        onClick={[Function]}>
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</ListGroupItem>
`;
